src/openrct2/world/Footpath.cpp

```diff
@@ -524,55 +524,51 @@ static void footpath_connect_corners(int32_t initialX, int32_t initialY, TileEle
         return;
 
     tileElement[0] = initialTileElement;
-    int32_t z = initialTileElement->base_height;
+    int32_t z = initialTileElement->GetBaseZ();
     for (int32_t initialDirection = 0; initialDirection < 4; initialDirection++)
     {
-        int32_t x = initialX;
-        int32_t y = initialY;
+        auto currentPos = footpathPos;
         int32_t direction = initialDirection;
 
-        x += CoordsDirectionDelta[direction].x;
-        y += CoordsDirectionDelta[direction].y;
-        tileElement[1] = footpath_connect_corners_get_neighbour(x, y, z, (1 << direction_reverse(direction)));
+        currentPos += CoordsDirectionDelta[direction];
+        tileElement[1] = footpath_connect_corners_get_neighbour({ currentPos, z }, (1 << direction_reverse(direction)));
         if (tileElement[1] == nullptr)
             continue;
 
         direction = (direction + 1) & 3;
-        x += CoordsDirectionDelta[direction].x;
-        y += CoordsDirectionDelta[direction].y;
-        tileElement[2] = footpath_connect_corners_get_neighbour(x, y, z, (1 << direction_reverse(direction)));
+        currentPos += CoordsDirectionDelta[direction];
+        tileElement[2] = footpath_connect_corners_get_neighbour({ currentPos, z }, (1 << direction_reverse(direction)));
         if (tileElement[2] == nullptr)
             continue;
 
         direction = (direction + 1) & 3;
-        x += CoordsDirectionDelta[direction].x;
-        y += CoordsDirectionDelta[direction].y;
+        currentPos += CoordsDirectionDelta[direction];
         // First check link to previous tile
-        tileElement[3] = footpath_connect_corners_get_neighbour(x, y, z, (1 << direction_reverse(direction)));
+        tileElement[3] = footpath_connect_corners_get_neighbour({ currentPos, z }, (1 << direction_reverse(direction)));
         if (tileElement[3] == nullptr)
             continue;
         // Second check link to initial tile
-        tileElement[3] = footpath_connect_corners_get_neighbour(x, y, z, (1 << ((direction + 1) & 3)));
+        tileElement[3] = footpath_connect_corners_get_neighbour({ currentPos, z }, (1 << ((direction + 1) & 3)));
         if (tileElement[3] == nullptr)
             continue;
 
         direction = (direction + 1) & 3;
         tileElement[3]->AsPath()->SetCorners(tileElement[3]->AsPath()->GetCorners() | (1 << (direction)));
-        map_invalidate_element({ x, y }, tileElement[3]);
+        map_invalidate_element(currentPos, tileElement[3]);
 
         direction = (direction - 1) & 3;
         tileElement[2]->AsPath()->SetCorners(tileElement[2]->AsPath()->GetCorners() | (1 << (direction)));
 
-        map_invalidate_element({ x, y }, tileElement[2]);
+        map_invalidate_element(currentPos, tileElement[2]);
```

Comment:
This and subsequent calls to map_invalidate_element are calling with the wrong position. (note this is a bug in the current code)

Reply:
I'm guessing this is because the coords here should be `footpathPos + delta` for each of their directions, correct? This is what I've used.

If so, shouldn't all the places doing `currentPos += delta` actually be doing `currentPos = footpathPos + delta`?

Reply:
@duncanspumpkin can you weigh on this? If it's not clear I can explain it better.

Reply:
Solving based on discussions on gitter.